src/main/kotlin/graphics/scenery/Camera.kt

```diff
@@ -244,6 +243,21 @@ open class Camera : Node("Camera") {
     @JvmOverloads fun getNodesForScreenSpacePosition(x: Int, y: Int,
                                                        ignoredObjects: List<Class<*>> = emptyList(),
                                                        debug: Boolean = false): Scene.RaycastResult {
+        val (worldPos, worldDir) = screenPointToRay(x, y)
+
+        val scene = getScene()
+        if(scene == null) {
+            logger.warn("No scene found for $this, returning empty list for raycast.")
+            return Scene.RaycastResult(emptyList(), worldPos, worldDir)
+        }
+
+        return scene.raycast(worldPos, worldDir, ignoredObjects, debug)
+    }
+
+    /**
+     * Returns (worldPos, worldDir)
```

Comment:
Nice refactoring, please expand the docstring a bit though, and add a test for `screenPointToRay` to codecov shuts up ^^

Reply:
done also added test for the method above